console/console-init/ui/src/modules/device/dialogs/CreateDevice/AddGateway.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React, { useState } from "react";
+import {
+  Title,
+  TextInput,
+  FormGroup,
+  Form,
+  Button,
+  FlexItem,
+  Flex,
+  Chip,
+  ChipGroup,
+  GridItem,
+  Grid
+} from "@patternfly/react-core";
+import { DeviceListAlert } from "modules/device/components";
+
+export interface IGatewayProps {
+  addedGateways: string[];
+  removeGateway: (id: string) => void;
+}
+
+export const AddGateway: React.FunctionComponent<IGatewayProps> = ({
+  addedGateways,
+  removeGateway
+}) => {
+  const [inputID, setInputID] = useState<string>("");
+  const [isValid, setIsValid] = useState<boolean>(false);
+
+  const addGateway = (deviceId: string) => {
+    /* Handle adding gateway */
+  };
+
+  const deviceIdRegExp = new RegExp(/^[a-z0-9]+$/i);
```

Comment:
Regex should contain capital letter as well and special characters like `. _ - { } ( ) # @`